### PR TITLE
Enable and fix CA1825 warnings

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable and fix CA1825 warnings

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/3331)

### CODE REVIEW

File: src/System.Windows.Forms.Design/tests/UnitTests/System/ComponentModel/Design/ComponentDesignerTests.cs [hunk 1]
```diff
@@ -153,7 +153,7 @@ public void ComponentDesigner_Children_GetWithValidHostInvalidResult_ReturnsExpe
 
         public static IEnumerable<object[]> Children_GetInvalidService_TestData()
         {
-            foreach (ICollection associatedComponents in new object[] { null, new object[0], new object[] { new Component() }})
+            foreach (ICollection associatedComponents in new object[] { null, Array.Empty<object>(), new object[] { new Component() }})
```
Comment (hughbe) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/ComponentModel/Design/ComponentDesignerTests.cs:156]: nit: do you think it is worth having the code analysis run on tests? Could condition it with `Condition="'$(IsTestProject)' != 'true'"` or something

Comment (gpetrou) [reply src/System.Windows.Forms.Design/tests/UnitTests/System/ComponentModel/Design/ComponentDesignerTests.cs:156]: Are you talking about all rules or the specific one only?

Comment (hughbe) [reply src/System.Windows.Forms.Design/tests/UnitTests/System/ComponentModel/Design/ComponentDesignerTests.cs:156]: I meant all ones

Comment (gpetrou) [reply src/System.Windows.Forms.Design/tests/UnitTests/System/ComponentModel/Design/ComponentDesignerTests.cs:156]: Some rules might be useful for tests as well, such as the dispose-related ones, but lets see what @RussKie suggests.

Comment (RussKie) [reply src/System.Windows.Forms.Design/tests/UnitTests/System/ComponentModel/Design/ComponentDesignerTests.cs:156]: We have different rulesets for [shipping](https://github.com/dotnet/winforms/blob/master/eng/Shipping.ruleset) and [non-shipping](https://github.com/dotnet/winforms/blob/master/eng/NonShipping.ruleset) code (our tests are "non-shipping" artifacts).
If there is a good reason to exclude a specific rule from being applied to the non-shipping code - we can do so. In this instance I'm not seeing a reason to exclude, do you?

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DataGridViewRowTests.cs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Licensed to the .NET Foundation under one or more agreements.
+﻿// Licensed to the .NET Foundation under one or more agreements.
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DataGridViewRowTests.cs:1]: [nit] This can be reverted

**Pull request merged**